modules/@apostrophecms/schema/index.js

```diff
@@ -1698,12 +1706,87 @@ module.exports = {
       },
 
       registerAllSchemas() {
-        _.each(self.apos.doc.managers, function (manager, type) {
-          self.register('doc', type, manager.schema);
-        });
-        _.each(self.apos.area.widgetManagers, function (manager, type) {
-          self.register('widget', type, manager.schema);
-        });
+        self.schemaPointers = {};
+        registerMetaType(self.apos.doc.managers, 'doc');
+        registerMetaType(self.apos.area.widgetManagers, 'widget');
+        function registerMetaType(managers, metaType) {
+          for (const [ type, manager ] of Object.entries(managers)) {
+            const schema = manager.schema;
+            self.register(metaType, type, schema);
+            const pointer = {
```

Comment:
All this pointer stuff allows us to resolve from relative field names to field ids and then the actual field object, without trusting the client.